### ISSUE TITLE
Are you accepting significant contributions?

### ISSUE THREAD

**Issue by douglasg14b** [bloomtom/DEnc#4]
I need to modify this library to provide greater flexibility for my use cases (More control over output naming, more control over bitrate/quality, only generating a manifest and not re-encoding...etc), and in doing so a lot of the "messy" bits would be cleaned up during this process (Such as the `GenerateDash()` method being broken up into appropriate responsibilities so it's manageable).

Before I do this on a fork instead of my own rendition of your work, I want to know if you are accepting significant contributions?

**Comment by bloomtom** [comment]
I've actually been unhappy with `GenerateDash` for some time now, and have wanted to split it into three methods for probing, converting, and dash manifest generation. Is this approximately what you wanted to do?

**Comment by douglasg14b** [comment]
Essentially, yes. Though I cannot say if it would be limited to three methods. The goal being to separate the concerns of `GenerateDash()` so that they can be called/configured independently of the other, better supporting a wider range of use cases.


If you have any requirements or guidelines you wish for me to adhere to, I'm happy to hear them. I'd hate to put in significant effort to have to throw it away if the pull request doesn't fit your goals.

<hr>

If you are fine with me "having at it" I can at least assure you that it's in good hands, I always strive to leave a codebase in better condition than when I found it.

Either way I have a need to make it more extendable for my use case.

**Comment by douglasg14b** [comment]
@bloomtom I am starting this immediately since I'm on a tight deadline, are you on a chat platform where it would be relatively easy to query you regarding certain design elements that are unclear?

**Comment by bloomtom** [comment]
As long as you have a clear idea for a new public API, I'm okay with heavy modification to meet it. I'd just like to review the new method signatures and workflow for a full convert + DASH process.

My email address is bloomtom@all8bits.com. Since GitHub lacks a PM feature please email me, and I'll reply with my Discord ID.

**Comment by douglasg14b** [comment]
Email sent.

Out of curiosity have you considered semantically naming your project so it's easier to discover? ie. `.Net Dash`, `DashSharp`, `Dash.Net` or some other common `.net` library naming scheme?

I stumbled on this by happenstance, and initially passed it off by name alone till I got around to looking at each repo I had listed. That might be a regular occurrence for others as well.

Just my $0.02 on that, not really my purview as this is your project not mine.
